Orange/widgets/unsupervised/owmds.py

```diff
@@ -326,40 +249,49 @@ def button(action):
                         checkbox_label="Send selected automatically",
                         box=None)
 
-        self.plot = pg.PlotWidget(background="w", enableMenu=False)
         self.plot.setAspectLocked(True)
         self.plot.getPlotItem().hideAxis("bottom")
         self.plot.getPlotItem().hideAxis("left")
         self.plot.getPlotItem().hideButtons()
         self.plot.setRenderHint(QPainter.Antialiasing)
-        self.mainArea.layout().addWidget(self.plot)
-
-        self.selection_tool = PlotSelectionTool(parent=self)
-        self.zoom_tool = PlotZoomTool(parent=self)
-        self.pan_tool = PlotPanTool(parent=self)
-        self.pinch_tool = PlotPinchZoomTool(parent=self)
-        self.pinch_tool.setViewBox(self.plot.getViewBox())
-        self.selection_tool.setViewBox(self.plot.getViewBox())
-        self.selection_tool.selectionFinished.connect(self.__selection_end)
-        self.current_tool = self.selection_tool
-
-        def activate_tool(action):
-            self.current_tool.setViewBox(None)
-
-            if action is action_select:
-                active, cur = self.selection_tool, Qt.ArrowCursor
-            elif action is action_zoom:
-                active, cur = self.zoom_tool, Qt.ArrowCursor
-            elif action is action_pan:
-                active, cur = self.pan_tool, Qt.OpenHandCursor
-            self.current_tool = active
-            self.current_tool.setViewBox(self.plot.getViewBox())
-            self.plot.getViewBox().setCursor(QCursor(cur))
-
-        group.triggered[QAction].connect(activate_tool)
 
         self._initialize()
 
+    def reset_graph_data(self, *_):
+        if self.data is not None:
+            self.graph.rescale_data()
+            self.update_graph()
+
+    def update_colors(self):
+        pass
+
+    def update_density(self):
+        self.update_graph(reset_view=False)
+
+    def update_regression_line(self):
+        self.update_graph(reset_view=False)
+
+    def init_attr_values(self):
+        domain = self.data and len(self.data) and self.data.domain or None
+        for model in self.models:
+            model.set_domain(domain)
+        self.graph.attr_color = domain and self.data.domain.class_var or None
+        self.graph.attr_shape = None
+        self.graph.attr_size = None
+        self.graph.attr_label = None
+
+    def prepare_data(self):
+        pass
+
+    def update_graph(self, reset_view=True, **_):
+        self.graph.zoomStack = []
+        if self.graph.data is None:
+            return
+        self.graph.update_data(self.variable_x, self.variable_y, reset_view)
+
+    def selection_changed(self):
+        self.commit()
+
```

Comment:
Doesn't all this same code appear also in the scatter plot widget? @astaric suggested that this code belongs to `OWScatterPlotGraph`.

Reply:
Yes. The same code appears there. So can I make a super class from which MDS and Scatter Plot are inherited?